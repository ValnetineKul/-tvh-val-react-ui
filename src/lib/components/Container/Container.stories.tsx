import type { ComponentProps } from 'react';
import React from 'react';
import { createMeta } from '../../story-utils';
import Typography from '../Typography';
import Surface from '../Surfaces/Surface';
import Container from './Container';

type Props = ComponentProps<typeof Container>;
const Template = (args: Props) => <Container {...args} />;

export const Primitive = Template.bind({});
Primitive.args = {
  children: (
    <Surface color="150" style={{ height: '50vh' }}>
      <Typography variant="body500">Container content</Typography>
    </Surface>
  ),
  disableGutters: false,
};

Primitive.storyName = 'Container';

export default createMeta({
  component: Container,
  title: 'Foundations/Container',
  parameters: {
    layout: 'fullscreen',
  },
});
