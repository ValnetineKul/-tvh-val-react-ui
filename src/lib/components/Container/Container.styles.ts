import { makeStyles } from '../../../themes/core';

const useStyles = makeStyles<void, 'disableGutters'>({ name: 'Container' })((theme, _params, classes) => ({
  root: {
    maxWidth: 1272,
    paddingLeft: theme.layoutSpacing['spacing/400'],
    paddingRight: theme.layoutSpacing['spacing/400'],
    [theme.breakpoints.up('sm')]: {
      paddingLeft: theme.layoutSpacing['spacing/450'],
      paddingRight: theme.layoutSpacing['spacing/450'],
    },

    [`&.${classes.disableGutters}`]: {
      paddingLeft: theme.layoutSpacing['spacing/000'],
      paddingRight: theme.layoutSpacing['spacing/000'],
    },
  },
  disableGutters: {},
}));

export default useStyles;
