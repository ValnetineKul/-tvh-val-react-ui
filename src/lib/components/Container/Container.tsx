import type { FC } from 'react';
import React from 'react';
import { default as MuiContainer } from '@mui/material/Container';

import useStyles from './Container.styles';

export interface ContainerProps {
  children: React.ReactNode;
  className?: string;
  disableGutters?: boolean;
}

const Container: FC<ContainerProps> = ({ children, className, disableGutters }) => {
  const { classes, cx } = useStyles();
  return (
    <MuiContainer className={cx(classes.root, { [classes.disableGutters]: disableGutters }, className)}>
      <>{children}</>
    </MuiContainer>
  );
};

export default Container;
