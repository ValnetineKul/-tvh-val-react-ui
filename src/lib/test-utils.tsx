import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
