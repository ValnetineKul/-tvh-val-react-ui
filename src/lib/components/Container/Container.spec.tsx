import React from 'react';
import { render } from '../../test-utils';
import Container from './Container';

describe('Container', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <Container>
        <div>children</div>
      </Container>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
