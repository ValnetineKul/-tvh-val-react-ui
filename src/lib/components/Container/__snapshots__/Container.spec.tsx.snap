// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Should render correctly 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg tss-3cvn6o-Container-root mui-1oqqzyl-MuiContainer-root"
>
  <div>
    children
  </div>
</div>
`;
